ferrocene/tools/self-test/src/compile.rs

```diff
@@ -191,23 +193,25 @@ impl ExpectedFiles {
                 .ok_or_else(|| Error::NonUtf8Path { path: entry.path() })?
                 .to_string();
 
-            if !currently_expected.remove(&*file_name) {
+            if !currently_expected.remove(file_name.as_str()) {
                 return Err(Error::UnexpectedCompilationArtifact {
-                    name: file_name.into(),
+                    name: file_name,
                     after_compiling: after_compiling.into(),
                 });
             }
         }
+
         let mut currently_expected = currently_expected.into_iter().collect::<Vec<_>>();
         currently_expected.sort();
-        for missing_file in currently_expected {
-            return Err(Error::MissingCompilationArtifact {
-                name: missing_file.into(),
+
+        if let Some(missing_file) = currently_expected.first() {
+            Err(Error::MissingCompilationArtifact {
+                name: missing_file.to_string(),
```

Comment:
did `.into()` fail

Reply:
It told me I needed to specify the type for some reason

Reply:
But in general I am unsure about using `into`, `to_owned` or `to_string`. And I am definitely not consistent. Probably `into` is best, since it is shortest.

Reply:
I prefer `to_owned` over `to_string`, since it is more clear what it does... the original data is already a string type.

This is not a suggestion to change this, just a nit you can ignore.